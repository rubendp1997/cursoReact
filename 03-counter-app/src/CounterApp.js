import React, {useState} from 'react';
import PropTypes from 'prop-types'
//rafcp sinppet

const CounterApp = ({value}) => {

    const [counter, setCounter]= useState(value);


    const handleAdd = (e)=>{
       // setCounter(counter+1);
       setCounter((c)=>c+1)
    }

    const handleReset = (e)=>{
        // setCounter(value);
        setCounter(value)
     }

     const handleSubstract = (e)=>{
        // setCounter(counter-1);
        setCounter((c)=>c-1)
     } 


return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        
        <button onClick={handleAdd}>+1</button>
        <button className= "buttonReset" onClick={handleReset}>Reset</button>
        <button onClick={handleSubstract}>-1</button>

        
    </>
);

}


CounterApp.prototypes= {
    value: PropTypes.number
}

export default CounterApp;