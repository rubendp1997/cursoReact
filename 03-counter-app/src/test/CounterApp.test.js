import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import PrimeraApp from '../CounterApp';
import CounterApp from '../CounterApp';

describe('first', () => {
    let wrapper;

    beforeEach (() => {
        wrapper = shallow( <CounterApp value= {10}/> );
    }

    );
    // test('Debe de mostrar el mensaje hola soy perro', () => {
    //     const saludo = 'Hola soy Goku';

    //     const { getByText } = render(<PrimeraApp saludo = {saludo}/>);

    //     expect ( getByText(saludo) ).toBeInTheDocument();
    // });

    test('Correct snpashot', () => {
      
        const counter = 100;
        const wrapper = shallow( <CounterApp value= {counter}/> );

        expect(wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar el valor de contador', () => {

        const counter = '200';
        const wrapper = shallow( <CounterApp value = {counter}/> );


        const textoParrafo = wrapper.find('h2').text();
        expect(textoParrafo.trim()).toBe(counter);
      
    });

    test('Debe incrementar contador con +1', () => {
      

        wrapper.find('button').at(0).simulate('click');
        const textoContador = wrapper.find('h2').text();
        console.log(textoContador);
        expect(textoContador).toBe('11');

    });

    test('Debe incrementar contador con -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const textoContador = wrapper.find('h2').text();

        expect(textoContador).toBe('9');

    });

    test('Debe resetear contador', () => {
        wrapper = shallow( <CounterApp value= {10}/> );
        wrapper.find('button').at(2).simulate('click');

        wrapper.find('button').at(1).simulate('click');
        const textoContador = wrapper.find('h2').text();

        expect(textoContador).toBe('10');

    });
    
    
    
    
});
