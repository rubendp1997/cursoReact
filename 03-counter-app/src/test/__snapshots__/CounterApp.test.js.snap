// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first Correct snpashot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    className="buttonReset"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
